### PR TITLE
pkg/openwsn: bump version

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

With the new version some OpenWSN modules are optional:
- openwsn_cjoin (and therefore opewnsn_coap)
- openwsn_6lo_fragmentation
- openwsn_icmpv6echo
- openwsn_crypto
- openwsn_udp (optional but kept as default)

New optional modules have been added:
- openwsn_iee802154e_security (link layer security)
- openwsn_adaptive_msf (MSF dynamic slot allocation)

OpenVisualizer needs to be updated to be compatible with the new
version.

There is one thing I'm unsure of in this PR, OpenWSN now defines there own `LOG_%` functions, this conflicts with the ones in RIOT, therefore I copied the `log.h` header and added `LOG_RIOT_%` definitions. I could also remove all logging from the pkg, but I still find it usefull to have when debugging.

### Testing procedure

 * Install/update  openvisualizer:
   * git clone -b develop-RIOT https://github.com/fjmolinas/openvisualizer.git
   * cd openvisualizer
   * pip2 install .

* Follow the test [README](https://github.com/fjmolinas/RIOT/blob/105c5b639c400d719737be18b9bc4367402736c6/tests/pkg_openwsn/README.md)

Otherwise, follow or trust this condensed version of the README:

<details><summary><b>step by step test procedure</b></summary>

1. Setup an iotlab experiment

```
# book experiment and get nodes list
$ iotlab-experiment submit -d 120 -l 3,archi=m3:at86rf231+site=saclay
{
    "id": 226328
}
$ iotlab-experiment get --nodes
{
    "items": [
        {
            "archi": "m3:at86rf231",
            "camera": "0",
            "mobile": "0",
            "mobility_type": " ",
            "network_address": "m3-10.saclay.iot-lab.info",
            "production": "YES",
            "site": "saclay",
            "state": "Alive",
            "uid": "4061",
            "x": "1",
            "y": "58.8",
            "z": "4"
        },
        {
            "archi": "m3:at86rf231",
            "camera": "0",
            "mobile": "0",
            "mobility_type": " ",
            "network_address": "m3-11.saclay.iot-lab.info",
            "production": "YES",
            "site": "saclay",
            "state": "Alive",
            "uid": "c471",
            "x": "1.8",
            "y": "58.8",
            "z": "4"
        },
        {
            "archi": "m3:at86rf231",
            "camera": "0",
            "mobile": "0",
            "mobility_type": " ",
            "network_address": "m3-12.saclay.iot-lab.info",
            "production": "YES",
            "site": "saclay",
            "state": "Alive",
            "uid": "3560",
            "x": "2.6",
            "y": "58.8",
            "z": "4"
        }
    ]
}
```

2.. OPTIONAL: setup up parameters to speed up sync

```
export CFLAGS="-DEB_PORTION=4 -DSLOTFRAME_LENGTH=51 -DIEEE802154E_SINGLE_CHANNEL=17"
```

3. flash root node and start openv

```
$IOTLAB_NODE=m3-10.saclay.iot-lab.info USEMODULE=openwsn_serial make -C tests/pkg_openwsn/ flash openv-termroot
...
13:44:13 [OpenVisualizerServer:INFO] Setting DAG root...
13:44:15 [OpenVisualizerServer:SUCCESS] Setting mote 0f87 as root
13:44:15 [OpenVisualizerServer:INFO] starting RPC server
13:44:15 [RPL:INFO] registering DAGroot 52-bd-fc-65-10-6b-11-14
```

4. flash the other two nodes as leaf nodes

```
$ IOTLAB_NODE=m3-11.saclay.iot-lab.info make -C tests/pkg_openwsn/ flash term
$ IOTLAB_NODE=m3-12.saclay.iot-lab.info make -C tests/pkg_openwsn/ flash term
```

5. wait for them to synchronize and join (use the `ifconfig` command for this), for them to be able to send messages between each other DIO must be received on the root for both nodes:

```
13:59:13 [JRC:VERBOSE] New node: 56ad:fc65:106b:1114. Creating new OSCORE context in /home/francisco/workspace/RIOT3/tests/pkg_openwsn/bin/oscore_context_56adfc65106b1114.json.
13:59:13 [JRC:VERBOSE] received JRC join request
13:59:18 [JRC:VERBOSE] New node: 26bc:fb65:106b:1114. Creating new OSCORE context in /home/francisco/workspace/RIOT3/tests/pkg_openwsn/bin/oscore_context_26bcfb65106b1114.json.
13:59:18 [JRC:VERBOSE] received JRC join request
13:59:25 [RPL:INFO] received RPL DAO from bbbb:0:0:0:56ad:fc65:106b:1114
        - parents:
           bbbb:0:0:0:52bd:fc65:106b:1114
        - children:
14:01:18 [RPL:INFO] received RPL DAO from bbbb:0:0:0:56ad:fc65:106b:1114
        - parents:
           bbbb:0:0:0:52bd:fc65:106b:1114
        - children:
           bbbb:0:0:0:26bc:fb65:106b:1114
```

6. start a udp server on each leaf node, and send udp messages from one node to the other.

```
  ifconfig
Iface  2        HWaddr: 0B:97  NID: CA:FE

                Long HWaddr: 56:AD:FC:65:10:6B:11:14
                inet6 addr: bbbb::56ad:fc65:106b:1114

                IEEE802154E sync: 1
                6TiSCH joined: 1

                RPL rank: 2816
                RPL parent: 52:BD:FC:65:10:6B:11:14
                RPL children:
                        26:BC:FB:65:10:6B:11:14
                RPL DODAG ID: bbbb::52bd:fc65:106b:1114
> udp server start 8888
udp server start 8888
Set UDP server port to 8888
> udp send bbbb::26bc:fb65:106b:1114 8888 hello
udp send bbbb::26bc:fb65:106b:1114 8888 hello
Send 5 byte over UDP to [bbbb::26bc:fb65:106b:1114]:8888
msg.l2_sendDoneError: 0
Send success
> Received 12 bytes on port 8888
00000000  33  29  00  00  00  00  00  68  65  6C  6C  6F                  3).....hello
```

```
ifconfig
Iface  2        HWaddr: 7D:86  NID: CA:FE

                Long HWaddr: 26:BC:FB:65:10:6B:11:14
                inet6 addr: bbbb::26bc:fb65:106b:1114

                IEEE802154E sync: 1
                6TiSCH joined: 1

                RPL rank: 2816
                RPL parent: 52:BD:FC:65:10:6B:11:14
                RPL children:
                        56:AD:FC:65:10:6B:11:14
                RPL DODAG ID: bbbb::52bd:fc65:106b:1114
> udp server start 8888
udp server start 8888
Set UDP server port to 8888
> Received 12 bytes on port 8888
00000000  47  26  00  00  00  00  00  68  65  6C  6C  6F                  G&.....hello
udp send bbbb::56ad:fc65:106b:1114 8888 hello
udp send bbbb::56ad:fc65:106b:1114 8888 hello
Send 5 byte over UDP to [bbbb::56ad:fc65:106b:1114]:8888
msg.l2_sendDoneError: 0
Send success
>
```

7. repeat but without `openwsn_cjoin` included, joining will happen faster, other than that everything else should be the same.

Here is the logs of udp messages being sent without using cjoin  `USEMODULE += openwsn_cjoin`

```
14:07:42 [RPL:INFO] received RPL DAO from bbbb:0:0:0:26bc:fb65:106b:1114
        - parents:
           bbbb:0:0:0:52bd:fc65:106b:1114
        - children:
           bbbb:0:0:0:56ad:fc65:106b:1114
14:07:58 [RPL:INFO] received RPL DAO from bbbb:0:0:0:26bc:fb65:106b:1114
        - parents:
           bbbb:0:0:0:52bd:fc65:106b:1114
        - children:
           bbbb:0:0:0:56ad:fc65:106b:1114
14:08:04 [RPL:INFO] received RPL DAO from bbbb:0:0:0:26bc:fb65:106b:1114
        - parents:
           bbbb:0:0:0:52bd:fc65:106b:1114
        - children:
           bbbb:0:0:0:56ad:fc65:106b:1114
14:09:46 [RPL:INFO] received RPL DAO from bbbb:0:0:0:56ad:fc65:106b:1114
        - parents:
           bbbb:0:0:0:52bd:fc65:106b:1114
        - children:
           bbbb:0:0:0:26bc:fb65:106b:1114
```

```
ifconfig
Iface  2        HWaddr: 7D:86  NID: CA:FE

                Long HWaddr: 26:BC:FB:65:10:6B:11:14
                inet6 addr: bbbb::26bc:fb65:106b:1114

                IEEE802154E sync: 1

                RPL rank: 2816
                RPL parent: 52:BD:FC:65:10:6B:11:14
                RPL children:
                        56:AD:FC:65:10:6B:11:14
                RPL DODAG ID: bbbb::52bd:fc65:106b:1114
> Received 12 bytes on port 8888
00000000  04  83  00  00  00  01  00  68  65  6C  6C  6F                  .......hello
udp send bbbb::56ad:fc65:106b:1114 8888 hello
udp send bbbb::56ad:fc65:106b:1114 8888 hello
Send 5 byte over UDP to [bbbb::56ad:fc65:106b:1114]:8888
msg.l2_sendDoneError: 0
Send success
>
```

```
ifconfig
Iface  2        HWaddr: 7D:86  NID: CA:FE

                Long HWaddr: 26:BC:FB:65:10:6B:11:14
                inet6 addr: bbbb::26bc:fb65:106b:1114

                IEEE802154E sync: 1

                RPL rank: 2816
                RPL parent: 52:BD:FC:65:10:6B:11:14
                RPL children:
                        56:AD:FC:65:10:6B:11:14
                RPL DODAG ID: bbbb::52bd:fc65:106b:1114
> Received 12 bytes on port 8888
00000000  04  83  00  00  00  01  00  68  65  6C  6C  6F                  .......hello
udp send bbbb::56ad:fc65:106b:1114 8888 hello
udp send bbbb::56ad:fc65:106b:1114 8888 hello
Send 5 byte over UDP to [bbbb::56ad:fc65:106b:1114]:8888
msg.l2_sendDoneError: 0
Send success
>

```

</details>

### CODE REVIEW

File: pkg/openwsn/include/openwsn_log.h [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * @file
+ * @brief       System logging header OpenWSN definitions
+ *
+ * This header implements the same functionality as log.h but avoiding
+ * macro redefinitions with OpenWSN LOG_% functions.
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef LOG_RIOT_H
```
Comment (kaspar030) [inline pkg/openwsn/include/openwsn_log.h:22]: this needs fixing (`OPENWSN_LOG_RIOT_H`) here and at the bottom

Comment (kaspar030) [reply pkg/openwsn/include/openwsn_log.h:22]: please squash right in

Comment (fjmolinas) [reply pkg/openwsn/include/openwsn_log.h]: fixed

**Comment by fjmolinas** [comment]
Thanks for the review!

**Pull request merged**